### ISSUE TITLE
Entity Listener : Inject a service containing entity manager

### ISSUE THREAD

**Issue by mehdi-ghezal** [doctrine/DoctrineBundle#300]
Hello,

With the following services configuration we get this exception :

```
ORMException: It's a requirement to specify a Metadata Driver and pass it to Doctrine\ORM\Configuration::setMetadataDriverImpl().
```

_Services configuration_

``` yml
services:
    acme.invoice.listener:
        class: %acme.invoice.listener.class%
        arguments:
            - "@acme.invoice.manager"
        tags:
            - { name: doctrine.orm.entity_listener }

    acme.invoice.manager:
        class: %acme.invoice.manager.class%
        arguments:
            - "@doctrine.orm.entity_manager"
```

I think that DoctrineBundle get all services related to my listener before Doctrine Initialization then the Entity Manager is not yet fully configure. I try to use lazy service but I have the same issue.

Hope this report can help you to make DoctrineBundle more awesome ;)

**Comment by vansante** [comment]
I am running into the same problem. Has this not been fixed yet?

**Comment by khasinski** [comment]
+1

**Comment by stepansib** [comment]
Still waiting for proper solution...

**Comment by Jocsonme** [comment]
Then, how to use it, help me, 274251086@qq.com

**Comment by grizzlylab** [comment]
+1

**Comment by Ocramius** [comment]
You can make the listener lazy, fwiw. If you do that, then you simply get rid of the problem.

@kimhemsoe would that satisfy this issue enough to close this?

**Comment by zhukovra** [comment]
@Ocramius making listener lazy isn't working for me.

**Comment by FriTOol** [comment]
+1

**Comment by stepansib** [comment]
+!

**Comment by AlexeyKosov** [comment]
@Ocramius It doesn't work...

**Comment by Ocramius** [comment]
Needs test case or example scenario.

**Comment by kimhemsoe** [comment]
There is an open PR which is trying to fix this issue but it need some love.

https://github.com/doctrine/DoctrineBundle/pull/443

On top that i am like 99% sure you can fix this with lazy services, if you make the right services lazy, ex not the listener but the listener's arguments.

**Comment by ThomasLandauer** [comment]
@kimhemsoe I marked the listener's argument as lazy, but I still get the error message.  
I followed the steps from here: https://symfony.com/doc/2.8/service_container/lazy_services.html  
How can I verify that the service is really lazy?

**Comment by paq85** [comment]
Lazy service helped but this issue costed me quite some time to figure out what's going on and to get to this github page.
I hope this gets improved some day.

**Comment by Notmarrco** [comment]
the PR https://github.com/doctrine/DoctrineBundle/pull/443 has been merged 

For me it works like a charm. I think the issue can be closed.

@ThomasLandauer now just add ```lazy: true``` in your listener tags :
```yaml
    tags:
            - { name: doctrine.orm.entity_listener, lazy: true } 
```

**Comment by mikeSimonson** [comment]
closed by #433

**Comment by Soullivaneuh** [comment]
Had the same issue. No need to tag as lazy for me. Just require `symfony/proxy-manager-bridge` and the entity managers services will be directly lazy loaded.

Issue solved like that. :+1: 

Ref: https://symfony.com/doc/current/service_container/lazy_services.html
